legal-api/src/legal_api/reports/report.py

```diff
@@ -198,9 +198,17 @@ def _get_template_filename(self):
         if ReportMeta.reports[self._report_key].get('hasDifferentTemplates', False):
             # Get template specific to legal type
             specific_template = ReportMeta.reports[self._report_key].get(self._business.legal_type, None)
-            # Fallback to default if specific template not found
-            file_name = specific_template['fileName'] if specific_template else \
-                ReportMeta.reports[self._report_key]['default']['fileName']
+            if self._business.legal_type == 'CP' and self._filing.filing_type == 'correction':
+                corrected_filing_id = self._filing.filing_json['filing'].get('correction', {}).get('correctedFilingId')
+                original_filing = Filing.find_by_id(corrected_filing_id)
+                if self._is_special_resolution_correction(
+                    self._business.legal_type, self._filing.filing_json['filing'], self._business, original_filing
+                ):
+                    file_name = 'specialResolutionCorrectionApplication'
```

Comment:
Follows the UX name 
![image](https://github.com/bcgov/lear/assets/3484109/d5f772ab-8416-4969-9a3b-3030b55b1e29)

Reply:
Had to do this, because a correction document is required for a correction - Special Resolution Correction and corrections of corrections that originate from a special resolution have a different document